src/main/scala/it/pagopa/interop/backendforfrontend/service/types/SelfcareClientTypes.scala

```diff
@@ -1,16 +1,56 @@
 package it.pagopa.interop.backendforfrontend.service.types
 
+import it.pagopa.interop.backendforfrontend.error.BFFErrors.MissingSelfcareFields
 import it.pagopa.interop.backendforfrontend.model._
+import it.pagopa.interop.commons.utils.TypeConversions.OptionOps
 import it.pagopa.interop.selfcare.v2.client.{model => SelfcareClient}
 
+import scala.concurrent.Future
+
 object SelfcareClientTypes {
 
   implicit class ProductResourceConverter(private val p: SelfcareClient.ProductResource) extends AnyVal {
-    def toApi: SelfcareProduct = SelfcareProduct(id = p.id, name = p.title)
+    def toApi: Future[SelfcareProduct] = {
+      val selfcareProduct = for {
+        id    <- p.id
+        title <- p.title
+      } yield SelfcareProduct(id = id, name = title)
+
+      selfcareProduct.toFuture {
+        val missingFields: String = List(
+          p.id.fold(Option("id"))(_ => Option.empty[String]),
+          p.title.fold(Option("title"))(_ => Option.empty[String])
+        ).flatten
+          .mkString(", ")
+        MissingSelfcareFields(p.getClass.getName, missingFields)
+      }
+    }
   }
 
   implicit class SelfcareInstitutionConverter(private val i: SelfcareClient.InstitutionResource) extends AnyVal {
-    def toApi: SelfcareInstitution =
-      SelfcareInstitution(id = i.id, description = i.description, userProductRoles = i.userProductRoles)
+    def toApi: Future[SelfcareInstitution] = {
+      val selfcareInstitution = for {
+        id               <- i.id
+        description      <- i.description
+        userProductRoles <- i.userProductRoles
+        parent           <- i.rootParent.map(_.description)
```

Comment:
`parent` is optional, but used in this for comprehension it is considered as required